flow/diff_context_unittests.cc

```diff
@@ -32,5 +32,35 @@ TEST_F(DiffContextTest, ClipAlignment) {
   EXPECT_EQ(damage.buffer_damage, SkIRect::MakeLTRB(16, 16, 64, 64));
 }
 
+#ifdef NOT_WORKING_YET
```

Comment:
@knopp here was my attempt to write a test case that triggered the above code, but it didn't seem to work (in fact it crashed a few times as I was working on it).

Reply:
I'll take a look at this first thing tomorrow.

Reply:
I actually fixed it. The reason I was having trouble reproducing it is that the test helper function was inserting a clip layer which effectively deleted the out-of-range layer. By "rolling my own" in the test case I was able to get the code that checks the return value to actually encounter a case where the frame/buffer damage was not empty, but needed to be set to empty and the test case triggers that condition now.

I wonder if there were any bugs in partial repaint caused by this? (i.e. something animating off screen and we kept repainting even though we didn't need to?)